ui/lib/kv/addon/routes/secret.js

```diff
@@ -35,6 +35,13 @@ export default class KvSecretRoute extends Route {
     return null;
   }
 
+  async isPatchAllowed(backend, path) {
+    if (!this.version.isEnterprise) return false;
+    const canPatch = await this.capabilities.canPatch(`${backend}/data/${path}`);
+    const canReadSubkeys = await this.capabilities.canRead(`${backend}/subkeys/${path}`);
```

Comment:
This is slightly inefficient because the second check doesn't kick off until the first one is finished -- not a blocker but something we can optimize later (possibly with `hash`?)

Reply:
Great suggestion! I refactored to this: 
```js
  async isPatchAllowed(backend, path) {
    if (!this.version.isEnterprise) return false;
    const capabilities = {
      canPatch: this.capabilities.canPatch(`${backend}/data/${path}`),
      canReadSubkeys: this.capabilities.canRead(`${backend}/subkeys/${path}`),
    };
    return hash(capabilities).then(
      ({ canPatch, canReadSubkeys }) => canPatch && canReadSubkeys,
      // if either promise fails return false
      () => false
    );
  }